### ISSUE TITLE
openmpi fails to build with intel compilers 18

### ISSUE THREAD

**Issue by alinelena** [open-mpi/ompi#4280]
Thank you for taking the time to submit an issue!

## Background information
I try to build openmpi 3.0.0 with intel  compilers 18.0 on a opensuse tumbleweed. this has gcc 7 as default.

configure stage fails with

```
In file included from a.c(67):
/usr/include/stdlib.h(133): error: identifier "_Float128" is undefined
  extern _Float128 strtof128 (const char *__restrict __nptr,
         ^

In file included from a.c(67):
/usr/include/stdlib.h(190): error: identifier "_Float128" is undefined
                        _Float128 __f)
                        ^

In file included from a.c(67):
/usr/include/stdlib.h(236): error: identifier "_Float128" is undefined
  extern _Float128 strtof128_l (const char *__restrict __nptr,
         ^

compilation aborted for a.c (code 2)
```

I have looked into the test and removing the following define , #define _GNU_SOURCE 1                                                                                                                                           , make it compile with icc without issues
I attached my config.log. I am still not sure if this is a failure of the build system of intel compiler.

Alin
[config.log](https://github.com/open-mpi/ompi/files/1341733/config.log)

**Comment by alinelena** [comment]
also glibc version is 2.26

**Comment by alinelena** [comment]
other woraround seems to be including stdio.h before the offending define

`
 #include <stdio.h>
 #define _GNU_SOURCE 1
 #include <stdlib.h>
 int
 main ()
 {
     long data[2] = {0, 0};
     long *lp;
     int *ip;
     ip = (int*) data;
     ip++;
     lp = (long*) ip;
     return lp[0];
   ;
   return 0;
 }
`

**Comment by rhc54** [comment]
I just downloaded/installed the Intel 2018 compilers and built OMPI master without problem. However, I am on a CentOS-7 box.

**Comment by kloeffelt** [comment]
I had the same problem, apparently it's an incompatibility between icc and gcc 7.0 / glibc 2.26 which will be fixed in Intel 18.0.1, here is the link to the Intel forum:
[https://software.intel.com/en-us/forums/intel-c-compiler/topic/742701](https://software.intel.com/en-us/forums/intel-c-compiler/topic/742701)

> yes indeed the problem persists in 18.0. I think Judy meant that the problem would be fixed in the first _update_ to 18.0 (i.e. 18.0.1). Anyway, for the time being I'm working around the problem by adding "-D_Float128=__float128" to my compiler flags.
Cheers, Bastian

Tobias

**Comment by rhc54** [comment]
Yeah, I didn't have gcc 7.0 installed on my box - using x.0 grade software is always risky. I'm going to close this as it isn't something we would address in OMPI - best to just wait for the updated compiler.
